lib/xebow.ex

```diff
@@ -37,4 +41,105 @@ defmodule Xebow do
 
   @spec layout() :: Layout.t()
   def layout, do: @layout
+
+  use GenServer
+
+  # Client Implementations:
+
+  @spec start_link([]) :: GenServer.on_start()
+  def start_link([]) do
+    GenServer.start_link(__MODULE__, Animation.types(), name: __MODULE__)
+  end
+
+  @doc """
+  Gets the current animation configuration. This retrievs current values, which
+  allows for changes to be made with `update_animation_config/2`
+  """
+  @spec get_animation_config() :: {Animation.Config.t(), keyword(Animation.Config.t())}
+  def get_animation_config do
+    GenServer.call(__MODULE__, :get_animation_config)
+  end
+
+  @doc """
+  Switches to the next active animation
+  """
+  @spec next_animation() :: :ok
+  def next_animation do
+    GenServer.cast(__MODULE__, :next_animation)
+  end
+
+  @doc """
+  Switches to the previous active animation
+  """
+  @spec previous_animation() :: :ok
+  def previous_animation do
+    GenServer.cast(__MODULE__, :previous_animation)
+  end
+
+  @doc """
+  Updates the animation configuration for the current animation
+  """
+  @spec update_animation_config(Animation.t()) :: :ok | :error
+  def update_animation_config(animation_with_config) do
+    GenServer.call(__MODULE__, {:update_animation_config, animation_with_config})
+  end
+
+  # Server Implementations:
+
+  @impl GenServer
+  def init(types) do
+    active_animations =
+      types
+      |> Enum.map(&initialize_animation/1)
+
+    [current | _] = active_animations
+    Engine.set_animation(current)
+    state = {active_animations, []}
```

Comment:
:information_source: (optional)

I recommend a state that's a map or struct because the keys are self-documenting. For example, it's not evident what `[]` is for unless I dig through the rest of this module.